classes/blocks/class-media.php

```diff
@@ -23,70 +23,75 @@ class Media extends Base_Block {
 	 */
 	const BLOCK_NAME = 'media_video';
 
-	/**
-	 * Register shortcake shortcode.
-	 *
-	 * @param array  $attributes Shortcode attributes.
-	 * @param string $content   Content.
-	 *
-	 * @return mixed
-	 */
-	public function add_block_shortcode( $attributes, $content ) {
-
-		$attributes = shortcode_atts(
-			[
-				'video_title'      => '',
-				'description'      => '',
-				'youtube_id'       => '',
-				'video_poster_img' => '',
-			],
-			$attributes,
-			'shortcake_media_video'
-		);
-
-		return $this->render( $attributes );
-	}
-
 	/**
 	 * Media constructor.
 	 */
 	public function __construct() {
-		add_shortcode( 'shortcake_media_video', [ $this, 'add_block_shortcode' ] );
-
 		register_block_type(
 			'planet4-blocks/media-video',
 			[
 				'editor_script'   => 'planet4-blocks',
-				'render_callback' => [ $this, 'render' ],
+				'render_callback' => static function ( $attributes, $content ) {
+					if ( '' !== trim( $content ) ) {
+						return $content;
+					}
+
+					$attributes = array_merge( $attributes, Media::get_media_data( $attributes ) );
+
+					if ( isset( $attributes['youtube_id'] ) ) {
+						$attributes['media_url'] = $attributes['youtube_id'];
+						unset( $attributes['youtube_id'] );
+					}
+
+					$json = wp_json_encode(
+						[ 'attributes' => $attributes ]
+					);
+
+					return '<div data-render="planet4-blocks/media-video" data-attributes="' . htmlspecialchars( $json ) . '"></div>';
```

Comment:
Since this will only run on old content, and so the logic needed is essentially "frozen", we might as well make it render the whole block in PHP. Then we could reduce the size of our front end bundle by not having to include the component code. Also page render time would benefit from less script execution/DOM manipulation.

We would need to write that logic in PHP anyway if we want to migrate old content to the new saved blocks.

But the current PR works so we can consider that in a follow up.